src/ucp/core/ucp_worker.c

```diff
@@ -2445,8 +2448,18 @@ static void ucp_worker_discard_uct_ep_cleanup(ucp_worker_h worker)
 
     kh_foreach(&worker->discard_uct_ep_hash, uct_ep, req, {
         ucs_assert(uct_ep == req->send.discard_uct_ep.uct_ep);
-        uct_ep_pending_purge(uct_ep, ucp_worker_discard_uct_ep_purge, NULL);
+        purged_req = NULL;
+        uct_ep_pending_purge(uct_ep, ucp_worker_discard_uct_ep_purge,
```

Comment:
/* Make sure UCP endpoint won't be destroyed by ucp_worker_discard_uct_ep_complete() since 
  ucp_request_t::send.comp.cb of another operation could access it */
- do add_ref before / remove_ref after

Reply:
done